File: python/1/part2.py
import pathlib

with open(pathlib.Path(__file__).parent.joinpath("input.txt"), "r", encoding="utf-8") as file:
    lines = file.readlines()

column_1 = []
column_2 = []

for line in lines:
    columns = line.split("   ")
    column_1.append(int(columns[0]))
    column_2.append(int(columns[1]))

column_1.sort()
column_2.sort()

cont_column = {}
after = None
for i in column_2:
    if i == after:
        continue
    else:
        cont_column[i] = column_2.count(i)
        after = i

result = 0
for j in column_1:
    try:
        result += cont_column[j] * j
    except KeyError:
        continue

print(result)

File: c++/1/part2.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <unordered_map>

int main() {
    std::ifstream file("input.txt");
    if (!file.is_open()) {
        std::cerr << "Can not open the file." << std::endl;
        return 1;
    }

    std::vector<int> column_1, column_2;
    std::string line;

    while (std::getline(file, line)) {
        std::istringstream iss(line);
        int num1, num2;
        if (iss >> num1 >> num2) {
            column_1.push_back(num1);
            column_2.push_back(num2);
        } else {
            std::cerr << "Error in line: " << line << std::endl;
        }
    }

    file.close();

    std::unordered_map<int, int> cont_column_2;
    for (int num : column_2) {
        cont_column_2[num]++;
    }

    int result = 0;
    for (int num : column_1) {
        if (cont_column_2.find(num) != cont_column_2.end()) {
            result += num * cont_column_2[num];
        }
    }

    std::cout << "Solution: " << result << std::endl;

    return 0;
}
